src/AI/SpenderAI.java

```diff
@@ -221,7 +223,7 @@ else if( unitQueue.isEmpty() )
             boolean validTarget = false;
             ArrayList<XYCoord> validTargets = new ArrayList<>();
 
-            if( unit.model.possibleActions.contains(UnitActionFactory.CAPTURE) )
+            if( uc.calculatePossibleActions().contains(UnitActionFactory.CAPTURE) )
```

Comment:
Would it be more idiomatic to call `unit.hasActionType(UnitActionFactory.CAPTURE)`?

Reply:
I guess it could incur extra overhead if you check for the presence of more than one action type (as Wally does), but it also feels annoying to have to create a `UnitContext` to learn about the `Unit` I already have.

Reply:
Probably more idiomatic, yes. I probably just wrote this before I wrote that.

It's definitely annoying and I don't think there's a solution under the current paradigm - the `Unit` itself is no help and the `UnitModel` is generic/sacrosanct.

I suppose another way of making this work would be to have modifiers return just the stat they're supposed to change... but then you run into the question of how you give them all the relevant information.
`UnitContext` also does allow for more complex interactions. Let's say I have a CO that lets his tanks capture, and another CO whose power grants a defense boost to units that can capture. You could handle that with the original setup, but immutable/shared models means that you have to provide the modifications at the time the other CO asks for them.